docs/3.x/tech/release-notes/index.adoc

```diff
@@ -12,14 +12,57 @@ Hide Stuff here
 
 // REMEMBER TO ADD MIGRATIONS TO upgrades-patches
 
+[role=release-note]
+== Version 3.31.3
+_Released Tuesday November 9th, 2021_
+
+=== Fixed
+* When adding an automated user action for a user action that is not time based you are unable to save.
+** Resolves https://github.com/CleanSpeak/cleanspeak-issues/issues/14[Github Issue #14]
+* Restoring from backup when the array `WhitelistedURLs` has an entry that is a `String` instead of an `Object` throws an exception.
```

Comment:
Isn't the bug when we restore from a backup and there are no tags for a particular entry? (when then affects which constructor Jackson wants to use)

Reply:
Actually, something like `"whitelistedURLs": [{ "url": "example.com", "tags": [] }]` won't cause an exception, but I could repro the exception using `"whitelistedURLs": ["stringOnlyEntry"]`.

Reply:
Hm.. `"whitelistedURLs": ["stringOnlyEntry"].` isn't valid, we would never see code like that - or if we did it would be a bug in our export. Unless we used to serialize white list URLs as such, we could look through git history I suppose. 

I think the proper repro is  (at least that is what I assumed based upon the Jackson error)

```
"whitelistedURLs": [ { 
   "url": "example.com"
 }]
```

Reply:
Before and currently `"whitelistedURLs": [{ "url": "example.com" }]` will not cause an exception.  The `tags` are defaulted to empty `Set`, so it's okay.